runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -232,12 +242,34 @@ private void addInteracting(int npcId)
 		interactedNpcIds.add(npcId);
 
 		// Add alternate forms of bosses
-		final Boss boss = Boss.getBoss(npcId);
+		final Boss boss = Boss.findBoss(npcId);
 		if (boss != null)
 		{
 			modifier = boss.getModifier();
-			interactedNpcIds.addAll(boss.getIds());
+			if (boss.isTob())
+			{
+				int partySize = getTobPartySize();
+				modifier = boss.getModifier(partySize);
+			}
+			interactedNpcIds.addAll(Sets.newHashSet(Arrays.stream(boss.getIds()).boxed().toArray(Integer[]::new)));
```

Comment:
```suggestion
			interactedNpcIds.addAll(Arrays.stream(boss.getIds()).boxed().collect(Collectors.toSet()));
```

Or just add the IDs in a loop:

```java
for (int bossId : boss.getIds())
{
	interactedNpcIds.add(bossId);
}
```